CMakeLists.txt

```diff
@@ -115,6 +116,10 @@ if (ENABLE_KVS_THREADPOOL)
   add_definitions(-DENABLE_KVS_THREADPOOL)
 endif()
 
+if (ENABLE_STATS_CALCULATION_CONTROL)
```

Comment:
Would it be helpful to include a note explaining that this modification is for memory optimization, along with an estimate of the memory savings if this feature were disabled?

Reply:
I have "Disabling these stats may lead to reductions in memory use." in the ReadMe. Will leave this comment open so we can consider adding a memory saving estimation for all of these, perhaps after more testing is done?